crates/bitwarden-crypto/src/keys/symmetric_crypto_key.rs

```diff
@@ -59,31 +59,35 @@ impl SymmetricCryptoKey {
         self.key.len() + self.mac_key.as_ref().map_or(0, |mac| mac.len())
     }
 
-    pub fn to_base64(&self) -> String {
-        let mut buf = self.to_vec();
-
-        let result = STANDARD.encode(&buf);
-        buf.zeroize();
-        result
+    pub fn to_base64(&self) -> SensitiveString {
+        self.to_vec().encode_base64(STANDARD)
     }
 
-    pub fn to_vec(&self) -> Zeroizing<Vec<u8>> {
-        let mut buf = Vec::with_capacity(self.total_len());
+    pub fn to_vec(&self) -> SensitiveVec {
+        let mut buf = SensitiveVec::new(Box::new(Vec::with_capacity(self.total_len())));
```

Comment:
Maybe add a comment why it's important to allocate the full length. It could be seen as an optimization method otherwise.
```suggestion
        // Prevent accidental copies by allocating the full size
        let mut buf = SensitiveVec::new(Box::new(Vec::with_capacity(self.total_len())));
```

Reply:
Done